subsys/bluetooth/mesh/pb_adv.c

```diff
@@ -412,12 +417,6 @@ static void gen_prov_start(struct prov_rx *rx, struct net_buf_simple *buf)
 		return;
 	}
 
-	if (rx->xact_id != expected_id) {
```

Comment:
This check has disappeared from this function completely. Should be reintroduced, along with a duplication check, for instance in combination with the check on line 411, like:

```C
if (rx->xact_id == link.rx.id) {
	if (!link.rx.seg) {
		if (!ack_pending()) {
			BT_DBG("Resending ack");
			gen_prov_ack_send(rx->xact_id);
		}

		return;
	}

	if (!(link.rx.seg & BIT(0))) {
		BT_DBG("Ignoring duplicate segment");
		return;
	}
} else if (rx->xact_id != next_transaction_id(link.rx.id)) {
	BT_WARN("Unexpected xact 0x%x, expected 0x%x", rx->xact_id,
		next_transaction_id(link.rx.id));
	return;
}
```